ledfx/api/websocket.py

```diff
@@ -298,21 +312,63 @@ def audio_stream_data_handler(self, message):
         )
 
 
+class UdpProtocol(asyncio.DatagramProtocol):
+    def __init__(self):
+        super().__init__()
+
+    def connection_made(self, transport):
+        self.transport = transport
+
+    def datagram_received(self, data, addr):
+        fdata = np.array(
+            [
+                (int.from_bytes(data[i : i + 2], "little") - 2048) / 2048
+                for i in range(0, len(data), 2)
+            ],
+            dtype=np.float32,
+        )
+        ACTIVE_AUDIO_STREAM.data = fdata
+
+
 class WebAudioStream:
-    def __init__(self, client: str, callback: callable):
+    def __init__(self, client: str, udp_port: int, callback: callable):
```

Comment:
Is this an extension of otherwise abandoned code?

Always nervous when I see new params, if its a dead stub of code though, makes more sense, though I need to look how this is generaly handled at a config level and if that is the proper pattern, just never been down in here before.

Reply:
What do you mean by abandoned? If in the sense that it hasn't been touched in a while, then yeah. As for if it's a proper pattern, I am not entirely sure. I'd like to implement a setting in configs which would allow the change of UDP port. I am very welcome to suggestions as to how I could go about it.

Reply:
I will need to go see where it is already used, I have some digging to do before I can offer more constructive input. I am assuming that adding a port number to the audio configuration via schema is relatively easy, the tricky bit will be only displaying or allowing edit for audio sources that use it.

Reply:
Note these are tourist statements on my part right now, I want to have a closer look on the weekend, but also assuming you are not in a rush to close out.

Reply:
No rush at all! I will also be looking more in depth on the weekend. Assuming Schema can be used, I will be deleting this implementation, because that sounds much better. I was considering Schema already, but the one exposed to websocket.py is BASE_MESSAGE_SCHEMA, for which adding UDP port would not make much sense. So either a new one would have to be added or it would have to be renamed, both of which I did not want to do without some additional input.

Reply:
We will likely need input from @YeonV on how best to add UDP edit field into audio source config for specific sources only...

Reply:
Removed these parameters and the udp_port from WebAudioStream class and added a global instance of a data class, which holds the parameters so that both WebAudioStream and WebsocketConnection could access it. Should also be able to modify it with Schema.

Reply:
Actually, now I am thinking, I could re add these parameters and use AUDIO_CONFIG_SCHEMA for the UDP port and provide it from effects/audio.py.